### PR TITLE
Accept strings or list of strings as error condition in metadata.yaml

### LINKED ISSUE #183: Parsing errors in external rules
`metadata.yaml` parsing errors
```
{"level":"info","time":"2021-01-08T13:33:06Z","message":"Parsing /rules-content/external/rules/master_defined_as_machinesets/MASTER_DEFINED_AS_MACHINESETS/metadata.yaml"}
{"level":"error","error":"yaml: unmarshal errors:\n  line 2: cannot unmarshal !!seq into string","time":"2021-01-08T13:33:06Z","message":"Error trying to parse rule in dir /rules-content/external/rules/master_defined_as_machinesets"}
...
{"level":"info","time":"2021-01-08T13:33:06Z","message":"Parsing /rules-content/external/rules/subnets_migration_failure_massive_egressip/SUBNETS_MIGRATION_FAILURE_MASSIVE_EGRESSIP/metadata.yaml"}
{"level":"error","error":"yaml: unmarshal errors:\n  line 2: cannot unmarshal !!seq into string","time":"2021-01-08T13:33:06Z","message":"Error trying to parse rule in dir /rules-content/external/rules/subnets_migration_failure_massive_egressip"}
```
Missing required files in external rules:
```
{"level":"error","error":"Missing required file: summary.md","time":"2021-01-08T13:33:06Z","message":"Error trying to parse rule in dir /rules-content/external/rules/unable_to_sync_storage"}
...
{"level":"error","error":"Missing required file: generic.md","time":"2021-01-08T13:33:06Z","message":"Error trying to parse rule in dir /rules-content/external/security/CVE_2020_8555_kubernetes"}

### PR DESCRIPTION
**Pull request by epapbak**
# Description

This is a partial temporary fix for issue #183.
With this PR, the when metadata.yaml file is parsed, it no allows condition field
to be either a string (for backward compatibility) or a sequence of strings (new
rules including multiple error keys).

Fixes #183 

## Type of change

- Bug fix
- Documentation update

## Testing steps

- manual testing of content-service and smart-proxy to verify output content
- UTs:
  - Added correct conditions field in existing rule contents test.
  - Added incorrect conditions in tests/content/bad_metadata_condition and corresponding UT

### CODE REVIEW

File: content/content.go [hunk 1]
```diff
@@ -65,6 +77,30 @@ func readFilesIntoFileContent(baseDir string, filelist []string) (map[string][]b
...
 }
 
+// transformMetadataCondition takes content of condition field in metadata.yaml
+// and transforms it into a simple string
+func transformMetadataCondition(parsedCondition *interface{}, errorContent *RuleErrorKeyContent) error {
+	switch (*parsedCondition).(type) {
+	case string:
+		errorContent.Metadata.Condition = (*parsedCondition).(string)
+		return nil
+	case []interface{}:
```
Comment (joselsegura) [inline content/content.go:87]: Does it work with `[]string` instead of `[]interface{}`? Probably if it is not an array of string we can stop parsing it

**Review by joselsegura — APPROVED** [review]
LGTM

Maybe it could be good to add a test case with an array of numbers instead of strings.

Remember to run `make godoc` to regenerate the literate documentation, please

File: content/content.go (hunk 1 above)
Comment (epapbak) [reply content/content.go:87]: Nope. There's no luck on that one, when unmarshalled It turns into a []interface and not directly into a []string

**Review by Bee-lee — APPROVED** [review]
Looks good overall! I guess the most acceptable usage of `interface{}` lol

Also, I just checked smart-proxy to be sure and we're not using the `condition` in the `RuleWithContentResponse` struct, which is the one that the smart proxy API returns, thus no need to worry about rendering in OCM or something. https://github.com/RedHatInsights/insights-results-smart-proxy/blob/master/server/server.go#L526

Thanks!

**Pull request merged**